packages/cli/src/deploy/resolveConfig.ts

```diff
@@ -22,18 +24,34 @@ export async function resolveConfig({
   readonly systems: readonly System[];
   readonly libraries: readonly Library[];
 }> {
-  const libraries = findLibraries(forgeOutDir).map((library): Library => {
-    // foundry/solc flattens artifacts, so we just use the path basename
-    const contractData = getContractData(path.basename(library.path), library.name, forgeOutDir);
-    return {
-      path: library.path,
-      name: library.name,
-      abi: contractData.abi,
-      prepareDeploy: createPrepareDeploy(contractData.bytecode, contractData.placeholders),
-      deployedBytecodeSize: contractData.deployedBytecodeSize,
-    };
+  const requirePath = await findUp("package.json");
+  if (!requirePath) throw new Error("Could not find package.json to import relative to.");
+  const require = createRequire(requirePath);
+
+  const moduleOutDirs = config.modules.flatMap((mod) => {
```

Comment:
maybe wrap this in a set so if we install multiple ERC20s, we only need to load the libs once?

```suggestion
  const moduleOutDirs = new Set(config.modules.flatMap((mod) => {
```

or ideally, we add support for multiple out dirs in `findLibraries`

Reply:
ah makes sense, I had actually thought of deduplicating but then forgot. Will try it withing findLibraries 👍

Reply:
https://github.com/latticexyz/mud/pull/3321/files

Let me know if this looks good to you 🙏